crates/type-system/src/ontology/entity_type/mod.rs

```diff
@@ -183,6 +189,13 @@ impl EntityTypeReference {
     }
 }
 
+impl From<&VersionedUri> for &EntityTypeReference {
+    fn from(uri: &VersionedUri) -> Self {
+        // SAFETY: Self is `repr(transparent)`
+        unsafe { &*(uri as *const VersionedUri).cast::<EntityTypeReference>() }
+    }
+}
+
```

Comment:
With this only being used once (in `link_mapping`) do you think it's worth adding `unsafe` to the crate?

Reply:
I think so, we hadn't disabled unsafe anyway, just didn't have any. And we both agree that really there should be a safe form of this operation if it's `repr(transparent)`